### ISSUE TITLE
Font rendering crash with some customs fonts.

### ISSUE THREAD

**Issue by codec-abc** [ryanisaacg/quicksilver#467]
Hi,

I am building a game ([depo here](https://github.com/RustyVolley/RustyVolleySrc)) with Quicksilver (version 0.3.5) and I noticed that depending on the font I use the program can panic at runtime.

For example, using this [font](https://github.com/RustyVolley/RustyVolleySrc/blob/340b15edc552223bb8e04ed11e906b93100d5aeb/static/font10.ttf) makes the program panic when trying to generate the text "Click to start!" with this font style:

```rust
FontStyle::new(64.0, Color {
                r: 0.0f32,
                g: 0.5f32,
                b: 0.4f32,
                a: 1.0f32,
})
```

The call stack I got is 
```
thread 'main' panicked at 'attempt to add with overflow', C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\graphics\font.rs:56:38
stack backtrace:
   0: std::sys::windows::backtrace::set_frames
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\sys\windows\backtrace\mod.rs:94
   1: std::sys::windows::backtrace::unwind_backtrace
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\sys\windows\backtrace\mod.rs:81
   2: std::sys_common::backtrace::_print
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\sys_common\backtrace.rs:70
   3: std::sys_common::backtrace::print
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\sys_common\backtrace.rs:58
   4: std::panicking::default_hook::{{closure}}
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:200
   5: std::panicking::default_hook
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:215
   6: std::panicking::rust_panic_with_hook
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:478
   7: std::panicking::continue_panic_fmt
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:385
   8: std::panicking::rust_begin_panic
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:312
   9: core::panicking::panic_fmt
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libcore\panicking.rs:85
  10: core::panicking::panic
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libcore\panicking.rs:49
  11: quicksilver::graphics::font::{{impl}}::render::{{closure}}
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\graphics\font.rs:56
  12: rusttype::rasterizer::rasterize<closure>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\rusttype-0.7.3\src\rasterizer.rs:366
  13: rusttype::PositionedGlyph::draw<closure>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\rusttype-0.7.3\src\lib.rs:1035
  14: quicksilver::graphics::font::Font::render
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\graphics\font.rs:52
  15: rustyvolley::home_menu_state::{{impl}}::draw_window_content::{{closure}}
             at D:\Workplace\rust\rustyvolley\src\home_menu_state.rs:83
  16: quicksilver::lifecycle::asset::Asset<quicksilver::graphics::font::Font>::execute_or<quicksilver::graphics::font::Font,closure,closure>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\lifecycle\asset.rs:37
  17: quicksilver::lifecycle::asset::Asset<quicksilver::graphics::font::Font>::execute<quicksilver::graphics::font::Font,closure>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\lifecycle\asset.rs:23
  18: rustyvolley::home_menu_state::{{impl}}::draw_window_content
             at D:\Workplace\rust\rustyvolley\src\home_menu_state.rs:79
  19: rustyvolley::state_manager::{{impl}}::draw
             at D:\Workplace\rust\rustyvolley\src\state_manager.rs:110
  20: quicksilver::lifecycle::application::Application<rustyvolley::state_manager::StateManager>::draw<rustyvolley::state_manager::StateManager>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\lifecycle\application.rs:60
  21: quicksilver::lifecycle::run::run_impl<rustyvolley::state_manager::StateManager,closure>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\lifecycle\run.rs:76
  22: quicksilver::lifecycle::run::run_with<rustyvolley::state_manager::StateManager,closure>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\lifecycle\run.rs:52
  23: quicksilver::lifecycle::run::run<rustyvolley::state_manager::StateManager>
             at C:\Users\cviot\.cargo\registry\src\github.com-1ecc6299db9ec823\quicksilver-0.3.5\src\lifecycle\run.rs:40
  24: rustyvolley::main
             at D:\Workplace\rust\rustyvolley\src\main.rs:25
  25: std::rt::lang_start::{{closure}}<()>
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\src\libstd\rt.rs:64
  26: std::rt::lang_start_internal::{{closure}}
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\rt.rs:49
  27: std::panicking::try::do_call<closure,i32>
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:297
  28: panic_unwind::__rust_maybe_catch_panic
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libpanic_unwind\lib.rs:92
  29: std::panicking::try
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panicking.rs:276
  30: std::panic::catch_unwind
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\panic.rs:388
  31: std::rt::lang_start_internal
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\/src\libstd\rt.rs:48
  32: std::rt::lang_start<()>
             at /rustc/daa53a52a2667533d5fe59bfcc5b8614b79c3d31\src\libstd\rt.rs:64
  33: main
  34: invoke_main
             at d:\agent\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  35: __scrt_common_main_seh
             at d:\agent\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  36: BaseThreadInitThunk
  37: RtlUserThreadStart
error: process didn't exit successfully: `target\debug\rustyvolley.exe` (exit code: 101)
```

**Comment by ryanisaacg** [comment]
I think your issue is solved on 0.3.6 (latest version is 0.3.7). Can you update and let me know if that resolves it?

By the way, I tried the web build of your game and it's great!

**Comment by codec-abc** [comment]
> I think your issue is solved on 0.3.6 (latest version is 0.3.7). Can you update and let me know if that resolves it?

Sorry about that, I should have tried the newer version before creating the issue. It is indeed it is fixed.

> By the way, I tried the web build of your game and it's great!

Thanks! But you are the real MVP here. Quicksilver make it pretty trivial to write game like that (Plus I just basically port other people code to Rust).
